libraries/chain/include/graphene/chain/operation_type_to_id.hpp

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) 2018 Abit More, and contributors.
+ *
+ * The MIT License
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+#pragma once
+
+#include <graphene/chain/protocol/operations.hpp>
+
+#include <boost/preprocessor/variadic/to_seq.hpp>
+#include <boost/preprocessor/seq/for_each_i.hpp>
+
+namespace graphene { namespace chain {
+
+template <typename OperationType>
+struct operation_type_id_from_operation_type
+{
+   static const int value = -1;
+};
+
+#define GRAPHENE_GENERATE_OP_TYPE_TO_ID_MAPPER(r, data, i, elem) \
+template <> \
+struct operation_type_id_from_operation_type<elem> \
+{ \
+   static const int value = i; \
+}; \
+
+
+BOOST_PP_SEQ_FOR_EACH_I( GRAPHENE_GENERATE_OP_TYPE_TO_ID_MAPPER, , BOOST_PP_VARIADIC_TO_SEQ( GRAPHENE_OPERATIONS_VARIADIC ) )
+
+#undef GRAPHENE_GENERATE_OP_TYPE_TO_ID_MAPPER
+
+template <typename Action>
+void operation_type_from_operation_id(const int operation_type_id, const Action& action)
```

Comment:
What does this do?

Reply:
Thins function maps operation_type_id to specific operation type (create_account_operation for example) and calls Action::operator()() for this type.

Reply:
I feel this is not the standard usage of operation id in the backend, could you clarify please @xeroc how that is handled?

Reply:
I've fixed that. It was clarified in other thread. 
We now use operation_type instead of operation_id or operation_type_id in this feature.